### PR TITLE
Expose command history

### LINKED ISSUE #172: Expose command history from the tree in Open MCT
**Is your feature request related to a problem? Please describe.**
It's sometimes useful to see which commands have been recently executed, and in particular for this list to be updated in real-time.

**Describe the solution you'd like**
Yamcs provides [an API](https://docs.yamcs.org/yamcs-http-api/commands/list-commands/) for retrieving a list of commands executed recently, as well as a subscription mechanism. We could expose a new telemetry point from the tree called "Command History" which exposes command history, and real-time command execution like telemetry. This would allow it to be displayed in a telemetry table. 

A note on implementation, this should be implemented in the same way that the have implemented Events. That said, we currently have [a single object provider](https://github.com/akhenry/openmct-yamcs/blob/0d25ba199cf8dfcccd6a94a6747fe1d8b1cf3e49/src/providers/object-provider.js), [realtime telemetry provider](https://github.com/akhenry/openmct-yamcs/blob/0d25ba199cf8dfcccd6a94a6747fe1d8b1cf3e49/src/providers/realtime-provider.js), and [historical telemetry provider](https://github.com/akhenry/openmct-yamcs/blob/master/src/providers/historical-telemetry-provider.js) for both telemetry _and_ events, with internal logic to differentiate between the different cases (eg. [here](https://github.com/akhenry/openmct-yamcs/blob/master/src/providers/object-provider.js#L53), [here](https://github.com/akhenry/openmct-yamcs/blob/master/src/providers/historical-telemetry-provider.js#L200), and [here](https://github.com/akhenry/openmct-yamcs/blob/0d25ba199cf8dfcccd6a94a6747fe1d8b1cf3e49/src/providers/realtime-provider.js#L193)). Adding command history into the mix is likely to cause some spaghetification of our code, so as part of this we should look at how we might separate out the telemetry, Events, and Command History specific logic.

### PR DESCRIPTION
**Pull request by ozyx**
Closes #172 #180 

### Describe your changes:

Adds a new telemetry object "Commands" which exposes YAMCS Command History as telemetry in Open MCT (similar to the way Events are exposed). Extracts logic specific to handling of commands / events to their own files in order to clean up the provider code a bit.

Also fixes an issue which was causing the Event WebSocket connection to fail to establish.

Unlike historical command data (which arrives as single, complete, objects) realtime subscription command data arrives in chunks. Below is the same command displayed from history and realtime subscription:

<sub>Historical</sub>
image
<sub>Realtime</sub>
image

As a first pass, each chunk will be treated as a row in the telemetry table. In the future, we may want to look at updating an existing row as these chunks stream in.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [ ] Testing instructions included in associated issue?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Commit messages meet standards?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/providers/messages.js [hunk 1]
```diff
@@ -30,7 +31,15 @@ function buildSubscribeMessages() {
         subscriptionMessages[objectType] = (subscriptionDetails) => {
             let message;
 
-            if (isEventOrAlarmType(objectType)) {
+            if (isEventType(objectType)) {
+                message = `{
+                    "type": "${dataType}",
+                    "id": "${subscriptionDetails.subscriptionId}",
+                    "options": {
+                        "instance": "${subscriptionDetails.instance}"
```
Comment (ozyx) [inline src/providers/messages.js:39]: removed "processor" from the subscribe message for events as yamcs was rejecting it.

Open network tab and navigate to 'Events' on our testing environment to see the constant disconnect/reconnect. This fixes it.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 dist/
 node_modules/
+.DS_Store
```
Comment (akhenry) [inline .gitignore:3]: I think there's a carriage return missing here. (I know, it's a bit pedantic)

File: src/plugin.js [hunk 3]
```diff
@@ -146,6 +146,12 @@ export default function installYamcsPlugin(configuration) {
             cssClass: "icon-generator-events"
         });
 
+        openmct.types.addType(OBJECT_TYPES.COMMANDS_OBJECT_TYPE, {
+            name: "Commands",
+            description: "To view command history",
+            cssClass: "icon-generator-events" // TODO: replace
```
Comment (akhenry) [inline src/plugin.js:152]: Please file a followup issue and assign to Charles and Rukmini.

File: src/providers/commands.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    return commandObject;
+}
+
+/**
+ * Convert raw command data from YAMCS to a format which
+ * can be consumed by Open MCT as telemetry.
+ * @param {Object} command
+ * @returns {Object} telemetryPoint
+ */
+export function commandToTelemetryPoint(command) {
```
Comment (akhenry) [inline src/providers/commands.js:149]: It's confusing, but in the nomenclature we use a telemetry point is the thing producing telemetry (called a "parameter" in Yamcs, a "channel" at JPL, and a "mnemonic" in other environments - everyone uses their own terminology). We refer to the telemetry values themselves as telemetry data (or datum singular). So maybe this should be `commandToTelemetryDatum`?

File: src/providers/messages.js (hunk 1 above)
Comment (akhenry) [reply src/providers/messages.js:39]: Yikes! Good catch. This has worked recently (and still works in our sim environment) so I wonder if this is a change in Yamcs itself?

File: src/utils.js [hunk 5]
```diff
@@ -44,7 +44,8 @@ const VALUE_EXTRACT_MAP = {
     'STRING': (value) => value.stringValue,
     'ENUMERATED': (value) => value.stringValue,
     'TIMESTAMP': (value) => value.stringValue,
-    'BOOLEAN': (value) => value.booleanValue
+    'BOOLEAN': (value) => value.booleanValue,
+    'BINARY': (value) => value.binaryValue
```
Comment (akhenry) [inline src/utils.js:48]: Intriguing...I wonder what this looks like in JSON? Base64?

**Review by akhenry — CHANGES_REQUESTED** [review]
Looks awesome, nice job. I've left a couple of small suggestions.

File: src/utils.js (hunk 5 above)
Comment (ozyx) [reply src/utils.js:48]: Yep, this comes in as a Base64 string. The YAMCS explorer actually decodes this and displays it as hex separated into 2-byte chunks. I attempted to do the same here, but for the sake of simplicity of a first-pass, went with just displaying the Base64 string.

File: src/providers/messages.js (hunk 1 above)
Comment (ozyx) [reply src/providers/messages.js:39]: I think this was due to a change in the Yamcs API. I referenced the events subscription API and noticed it did not contain the processor parameter.

File: src/utils.js (hunk 5 above)
Comment (akhenry) [reply src/utils.js:48]: Cool! And yeah, Base64 is fine for now. Tackling binary as hex is out of scope.

File: src/providers/commands.js (hunk 4 above)
Comment (ozyx) [reply src/providers/commands.js:149]: Sounds good to me. I was also dissatisfied with "point". We use it elsewhere as well so I can go ahead and clean this up.

File: .gitignore (hunk 2 above)
Comment (ozyx) [reply .gitignore:3]: As a pedant myself, I have no qualms with fixing this

Comment (jvigliotta) [reply .gitignore:3]: Missing a period in your comment.

**Review by jvigliotta — APPROVED** [review]
Looks great! Nice work!

File: src/plugin.js (hunk 3 above)
Comment (ozyx) [reply src/plugin.js:152]: @akhenry 

Follow-up issue is here: https://github.com/akhenry/openmct-yamcs/issues/186

But I think we may need to add Rukmini and Charles as collaborators to this repo? I'm unable to assign to them.

**Review by akhenry — APPROVED** [review]

**Pull request merged**